packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -13,117 +13,69 @@ export function escapeRegexCharacters(str) {
 }
 
 /**
- * Filter suggestions
- * @param suggestions
- * @param value
+ * Render the datalist suggestion items container
+ * @param props
  */
-export function getMatchingSuggestions(suggestions = [], value) {
-	if (!value) {
-		return suggestions;
-	}
-
-	const escapedValue = escapeRegexCharacters(value.trim());
-	const regex = new RegExp(escapedValue, 'i');
-	return suggestions.filter(item => regex.test(item));
+function defaultRenderDatalistItemContainer(props) {
+	return <div {...props} />;
 }
 
 /**
- * Render the datalist input
- * @param props
+ * Render an empty container
  */
-function renderDatalistInput(props) {
+function defaultRenderNoMatch() {
 	return (
-		<div className={theme['typeahead-input-icon']}>
-			<FormControl {...props} />
-			<div className={theme['dropdown-toggle']}>
-				<span className="caret" />
-			</div>
+		<div className={`${theme['items-container']} ${theme['no-result']}`}>
+			<span>No match.</span>
 		</div>
 	);
 }
 
 /**
- * Render the datalist suggestion items container
- * @param props
+ * Convert a datalist item as a value/label pair if needed
  */
-function defaultRenderDatalistItemContainer(props) {
-	return <div {...props} />;
+function getValueLabelPair(item) {
+	if (typeof item === 'object') {
+		return { [item.value]: item.label };
+	}
+	return { [item]: item };
 }
 
 /**
- * Render the datalist suggestion item
- * @param item
- * @param value
+ * Returns the value/label map from given items
  */
-function renderDatalistItem(item, { value }) {
-	let emphasisedText = [item];
-	if (value) {
-		emphasisedText = [];
-		const regex = new RegExp(escapeRegexCharacters(value), 'gi');
-		const matchedValues = item.match(regex);
-		const restValues = item.split(regex);
-
-		for (let i = 0; i < restValues.length; i += 1) {
-			emphasisedText.push(restValues[i]);
-			if (matchedValues[i]) {
-				emphasisedText.push(
-					<em className={theme['highlight-match']}>
-						{matchedValues[i]}
-					</em>,
-				);
-			}
-		}
+function getItemsMap(items) {
+	if (!items || !items.length) {
+		return {};
 	}
 
-	return (
-		<div className={classnames(theme.item, 'datalist-item')}>
-			{emphasisedText.map((val, index) => <span key={index}>{val}</span>)}
-		</div>
-	);
+	const itemsMap = {};
+	items.forEach(item => Object.assign(itemsMap, getValueLabelPair(item)));
+	return itemsMap;
 }
```

Comment:
Why not start by defining the `itemsMap`. Return it empty in the if case or return it filed on the other case.